### PR TITLE
update(ci): only sign releases

### PR DESCRIPTION
**Pull request by LucaGuerra**
**What type of PR is this?**

> Uncomment one (or more) `/kind <>` lines:

/kind feature

**Any specific area of the project related to this PR?**

> Uncomment one (or more) `/area <>` lines:

/area tests

**What this PR does / why we need it**:

This updates the CI/CD process so that:
1. only container images for releases are signed (and not main)
2. uses the env variable `COSIGN_YES` for cosign 2.0

cc @developer-guy 

**Which issue(s) this PR fixes**:

Fixes #

**Special notes for your reviewer**:

### CODE REVIEW

**Review by developer-guy — DISMISSED** [review]
not a maintainer but LGTM

File: .github/workflows/docker-image.yaml [hunk 1]
```diff
@@ -73,9 +77,17 @@ jobs:
             RELEASE=${{ inputs.release }}
             COMMIT=${{ inputs.commit }}
             BUILD_DATE=${{ inputs.build_date }}
-      
+
+      - name: Install Cosign
+        if: inputs.sign
+        uses: sigstore/cosign-installer@03d0fecf172873164a163bbc64bed0f3bf114ed7 # v3.4.0
```
Comment (cpanato) [inline .github/workflows/docker-image.yaml:83]: you can bump to `sigstore/cosign-installer@dd6b2e2b610a11fd73dd187a43d57cc1394e35f9 # v3.0.5` and fix the version comment which is not correct

File: .github/workflows/docker-image.yaml [hunk 2]
```diff
@@ -73,9 +77,17 @@ jobs:
...
+        with:
+          cosign-release: 'v2.0.2'
```
Comment (cpanato) [inline .github/workflows/docker-image.yaml:85]: you can drop this the action already install the latest one

File: .github/workflows/docker-image.yaml [hunk 3]
```diff
@@ -73,9 +77,17 @@ jobs:
...
+
       - name: Sign the images with GitHub OIDC Token
+        if: inputs.sign
```
Comment (cpanato) [inline .github/workflows/docker-image.yaml:88]:
```suggestion
        if: ${{ inputs.sign }}
```

**Review by cpanato — COMMENTED** [review]
some comments

File: .github/workflows/docker-image.yaml (hunk 3 above)
Comment (LucaGuerra) [reply .github/workflows/docker-image.yaml]: Thanks! Honest question: what is the difference?

**Review by FedeDP — DISMISSED** [review]
LGTM! Thanks!
/approve

**Review by FedeDP — APPROVED** [review]
/approve

**Pull request merged**